docs/source/accelerate/fsdp.md

```diff
@@ -119,13 +119,6 @@ trainer = SFTTrainer(
     train_dataset=train_dataset,
     eval_dataset=eval_dataset,
     peft_config=peft_config,
-    packing=data_args.packing,
-    dataset_kwargs={
-        "append_concat_token": data_args.append_concat_token,
-        "add_special_tokens": data_args.add_special_tokens,
-    },
-    dataset_text_field=data_args.dataset_text_field,
-    max_seq_length=data_args.max_seq_length,
```

Comment:
Here would be an example where arguments are removed from `SFTTrainer` but no equivalent arguments were added to `training_args`.

Reply:
Ok I see what you mean.

When you run the script with, for example, `--max_seq_length 123`, the value will no longer feed `data_args` but `training_args`. Everything happens behind the scenes when arguments are parsed.


```
python example.py --output_dir tmp --max_seq_length 123
```

Before:

```python
from dataclasses import dataclass
from typing import Optional
from transformers import HfArgumentParser, TrainingArguments

@dataclass
class DataTrainingArguments:
    dataset_name: Optional[str] = None
    max_seq_length: int = 512

if __name__ == "__main__":
    parser = HfArgumentParser((DataTrainingArguments, TrainingArguments))
    data_args, training_args = parser.parse_args_into_dataclasses()
    print(data_args.max_seq_length)  # 123
```

After:

```python
from dataclasses import dataclass
from typing import Optional
from transformers import HfArgumentParser
from trl import SFTConfig

@dataclass
class DataTrainingArguments:
    dataset_name: Optional[str] = None

if __name__ == "__main__":
    parser = HfArgumentParser((DataTrainingArguments, SFTConfig))
    data_args, training_args = parser.parse_args_into_dataclasses()
    print(training_args.max_seq_length)  # 123
```